src/backend/utils/parse-modules.ts

```diff
@@ -0,0 +1,141 @@
+import {classDecorators, componentMetadata} from '../../tree/decorators';
+import {functionName} from '../../utils/function-name';
+
+export const AUGURY_TOKEN_ID_METADATA_KEY = '__augury_token_id';
+
+declare const ng;
+
+const resolveNgModuleDecoratorConfig = (m) => {
+  if (m.decorators) {
+    return m.decorators.reduce((prev, curr, idx, decorators) =>
+      prev ? prev : decorators[idx].type.prototype.toString() === '@NgModule' ? decorators[idx].args[0] : null, null);
+  }
+
+  return Reflect.getMetadata('annotations', m).find(decorator => decorator.toString() === '@NgModule');
+};
+
+export const parseModules = (firstRootDebugElement: any): {[key: string]: any} => {
+  const bootstrappedModule = firstRootDebugElement.injector.get(ng.coreTokens.ApplicationRef)._injector.instance;
+  const [modules, moduleNames, tokenIdMap] = _parseModule(bootstrappedModule.constructor);
+  const serializableModules = {};
+  for (const m in modules) {
+    if (modules.hasOwnProperty(m)) {
+      serializableModules[modules[m].name] = modules[m];
+    }
+  }
+  return {
+    names: moduleNames,
+    configs: serializableModules,
+    tokenIdMap,
+  };
+};
+
+const parseModuleName = (m) => {
+  return m.ngModule ? m.ngModule.name : m.name || (m.constructor ? m.constructor.name : null);
+};
+
+const randomId = () => {
+  return Math.random().toString(36).substring(7);
```

Comment:
@rajinder-yadav Actually since these are all generated at the same time, I don't think using the date will help much. I have a map of all the ids though, so I've added some code to regenerate on pre-existence.